### PR TITLE
Refactoring testing of jpsreport

### PR DESCRIPTION
**Pull request by chraibi**
In this PR the tests for `jpsreport` are included in the testing machinery as it is the case for `jpscore`.

This is important so that future changes in `jpsreport` can be testes by CI.

### CODE REVIEW

File: systemtest/jpsreport_tests/method_A/01_method_A.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        Tin_exit = True
+    else:
+        logging.info('--> T_in: different dist -- p-value: {0:.4f}'.format(Tin_p_value))
+        Tin_exit = False
+
+    if cumPed_p_value > alpha:
+        logging.info('--> cum_Ped: same dist (fail to reject H0) -- p-value: {0:.4f}'.format(cumPed_p_value))
+        cumPed_exit = True
+    else:
+        logging.info('--> cum_Ped: different dist (reject H0) -- p-value: {0:.4f}'.format(cumPed_p_value))
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_A/01_method_A.py]: logging.error

File: systemtest/jpsreport_tests/method_A/01_method_A.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        cumPed_exit = False
+
+    success = (Tin_exit == True) and (cumPed_exit == True)
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_A/01_method_A.py]: no need to check for `== True`: 
```python    
success = Tin_exit and cumPed_exit
```

File: systemtest/jpsreport_tests/method_A/01_method_A.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    if cumPed_p_value > alpha:
+        logging.info('--> cum_Ped: same dist (fail to reject H0) -- p-value: {0:.4f}'.format(cumPed_p_value))
+        cumPed_exit = True
+    else:
+        logging.info('--> cum_Ped: different dist (reject H0) -- p-value: {0:.4f}'.format(cumPed_p_value))
+        cumPed_exit = False
+
+    success = (Tin_exit == True) and (cumPed_exit == True)        
+    if not success:
+        logging.info("%s exits with FAILURE" % (argv[0]))
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_A/01_method_A.py]: logging.error

File: systemtest/jpsreport_tests/method_B/01_method_B.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_B/01_method_B.py]: Please add short documentation

File: systemtest/jpsreport_tests/method_B/01_method_B.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        density_exit = False
+
+    if velocity_p_value > alpha:
+        logging.info('--> velocity: same dist -- p-value: {0:.4f}'.format(velocity_p_value))
+        velocity_exit = True
+    else:
+        logging.info('--> velocity: different dist -- p-value: {0:.4f}'.format(velocity_p_value))
+        velocity_exit = False
+
+    success = (PedID_exit == True) and (velocity_exit == True) and (density_exit == True)
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_B/01_method_B.py]: see above, check for ==True not needed

File: systemtest/jpsreport_tests/method_B/01_method_B.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        PedID_exit = True
+    else:
+        logging.info('--> PedID: different dist -- p-value: {0:.4f}'.format(PedID_p_value))
+        PedID_exit = False
+
+    if density_p_value > alpha:
+        logging.info('--> density: same dist -- p-value: {0:.4f}'.format(density_p_value))
+        density_exit = True
+    else:
+        logging.info('--> density: different dist -- p-value: {0:.4f}'.format(density_p_value))
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_B/01_method_B.py]: logging.error

File: systemtest/jpsreport_tests/method_B/01_method_B.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    if velocity_p_value > alpha:
+        logging.info('--> velocity: same dist -- p-value: {0:.4f}'.format(velocity_p_value))
+        velocity_exit = True
+    else:
+        logging.info('--> velocity: different dist -- p-value: {0:.4f}'.format(velocity_p_value))
+        velocity_exit = False
+
+    success = (PedID_exit == True) and (velocity_exit == True) and (density_exit == True)
+    if not success:
+        logging.info("%s exits with FAILURE" % (argv[0]))
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_B/01_method_B.py]: logging.error

File: systemtest/jpsreport_tests/method_B/01_method_B.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+    density_ks_statistic, density_p_value = ks_2samp(data_2[:,1],data_1[:,1])
+
+    velocity_ks_statistic, velocity_p_value = ks_2samp(data_2[:,2],data_1[:,2])
+
+    if PedID_p_value > alpha:
+        logging.info('--> PedID: same dist -- p-value: {0:.4f}'.format(PedID_p_value))
+        PedID_exit = True
+    else:
+        logging.info('--> PedID: different dist -- p-value: {0:.4f}'.format(PedID_p_value))
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_B/01_method_B.py]: logging.error

File: systemtest/jpsreport_tests/method_A/01_method_A.py [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    Tin_ks_statistic, Tin_p_value = ks_2samp(data_2[:, 0],
+                                             data_1[:, 0])
+    cumPed_ks_statistic, cumPed_p_value = ks_2samp(data_2[:, 1],
+                                                   data_1[:, 1])
+
+    if Tin_p_value > alpha:
+        logging.info('--> T_in: same dist  -- p-value: {0:.4f}'.format(Tin_p_value))
+        Tin_exit = True
+    else:
+        logging.info('--> T_in: different dist -- p-value: {0:.4f}'.format(Tin_p_value))
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_A/01_method_A.py]: logging.error

File: systemtest/jpsreport_tests/method_C/01_method_C.py [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+    if density_p_value > alpha:
+        logging.info('--> density: same dist -- p-value: {0:.4f}'.format(density_p_value))
+        density_exit = True
+    else:
+        logging.info('--> density: different dist -- p-value: {0:.4f}'.format(density_p_value))
+        density_exit = False
+
+
+    success = (velocity_exit == True) and (density_exit == True)
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_C/01_method_C.py]: see above, == True not needed

File: systemtest/jpsreport_tests/method_C/01_method_C.py [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+    if not success:
+        logging.info("%s exits with FAILURE" % (argv[0]))
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_C/01_method_C.py]: logging.error

File: systemtest/jpsreport_tests/method_D/01_method_D.py [hunk 12]
```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_D/01_method_D.py]: Documentation

File: systemtest/jpsreport_tests/method_D/01_method_D.py [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+                                                     data_1[:, 5])
+    velocity_ks_statistic, velocity_p_value = ks_2samp(data_2[:, 6],
+                                                       data_1[:, 6])
+
+    if Frame_p_value > alpha:
+        logging.info('--> Frame: same dist -- p-value: {0:.4f}'.
+                     format(Frame_p_value))
+        Frame_exit = True
+    else:
+        logging.info('--> Frame: different dist -- p-value: {0:.4f}'.
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_D/01_method_D.py]: logging.error

File: systemtest/jpsreport_tests/method_D/01_method_D.py [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+                     format(Frame_p_value))
+        Frame_exit = False
+
+    if PersID_p_value > alpha:
+        logging.info('--> PersID: same dist -- p-value: {0:.4f}'.
+                     format(PersID_p_value))
+        PersID_exit = True
+    else:
+        logging.info('--> PersID: different dist -- p-value: {0:.4f}'.
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_D/01_method_D.py]: logging.error

File: systemtest/jpsreport_tests/method_D/01_method_D.py [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+                     format(PersID_p_value))
+        PersID_exit = False
+        
+    if x_p_value > alpha:
+        logging.info('--> x-coordinate: same dist -- p-value: {0:.4f}'.
+                     format(x_p_value))
+        x_exit = True
+    else:
+        logging.info('--> x-coordinate: different dist -- p-value: {0:.4f}'.
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_D/01_method_D.py]: logging.error

File: systemtest/jpsreport_tests/method_D/01_method_D.py [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+                     format(x_p_value))
+        x_exit = False
+        
+    if y_p_value > alpha:
+        logging.info('--> y-coordinate: same dist -- p-value: {0:.4f}'.
+                     format(y_p_value))
+        y_exit = True
+    else:
+        logging.info('--> y-coordinate: different dist -- p-value: {0:.4f}'.
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_D/01_method_D.py]: logging.error

File: systemtest/jpsreport_tests/method_D/01_method_D.py [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+                     format(y_p_value))
+        y_exit = False
+        
+    if z_p_value > alpha:
+        logging.info('--> z-coordinate: same dist -- p-value: {0:.4f}'.
+                     format(z_p_value))
+        z_exit = True
+    else:
+        logging.info('--> z-coordinate: different dist -- p-value: {0:.4f}'.
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_D/01_method_D.py]: logging.error

File: systemtest/jpsreport_tests/method_D/01_method_D.py [hunk 18]
```diff
@@ -0,0 +1,123 @@
...
+                     format(z_p_value))
+        z_exit = False
+        
+    if density_p_value > alpha:
+        logging.info('--> density: same dist -- p-value: {0:.4f}'.
+                     format(density_p_value))
+        density_exit = True
+    else:
+        logging.info('--> density: different dist -- p-value: {0:.4f}'.
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_D/01_method_D.py]: logging.error

File: systemtest/jpsreport_tests/method_D/01_method_D.py [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+                     format(density_p_value))
+        density_exit = False
+
+    if velocity_p_value > alpha:
+        logging.info('--> velocity: same dist -- p-value: {0:.4f}'.
+                     format(velocity_p_value))
+        velocity_exit = True
+    else:
+        logging.info('--> velocity: different dist -- p-value: {0:.4f}'.
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_D/01_method_D.py]: logging.error

File: systemtest/jpsreport_tests/method_D/01_method_D.py [hunk 20]
```diff
@@ -0,0 +1,123 @@
...
+                     format(velocity_p_value))
+        velocity_exit = False
+
+
+    success = (Frame_exit == True) and (PersID_exit == True) and \
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_D/01_method_D.py]: see above, no check for ==True needed

File: systemtest/jpsreport_tests/method_I/01_method_I.py [hunk 21]
```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_I/01_method_I.py]: Documentation

File: systemtest/jpsreport_tests/method_I/01_method_I.py [hunk 22]
```diff
@@ -0,0 +1,123 @@
...
+                                                     data_1[:, 5])
+    velocity_ks_statistic, velocity_p_value = ks_2samp(data_2[:, 6],
+                                                       data_1[:, 6])
+
+    if Frame_p_value > alpha:
+        logging.info('--> Frame: same dist -- p-value: {0:.4f}'.
+                     format(Frame_p_value))
+        Frame_exit = True
+    else:
+        logging.info('--> Frame: different dist -- p-value: {0:.4f}'.
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_I/01_method_I.py]: logging.error

File: systemtest/jpsreport_tests/method_I/01_method_I.py [hunk 23]
```diff
@@ -0,0 +1,123 @@
...
+                     format(Frame_p_value))
+        Frame_exit = False
+
+    if PersID_p_value > alpha:
+        logging.info('--> PersID: same dist -- p-value: {0:.4f}'.
+                     format(PersID_p_value))
+        PersID_exit = True
+    else:
+        logging.info('--> PersID: different dist -- p-value: {0:.4f}'.
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_I/01_method_I.py]: logging.error

File: systemtest/jpsreport_tests/method_I/01_method_I.py [hunk 24]
```diff
@@ -0,0 +1,123 @@
...
+                     format(PersID_p_value))
+        PersID_exit = False
+        
+    if x_p_value > alpha:
+        logging.info('--> x-coordinate: same dist -- p-value: {0:.4f}'.
+                     format(x_p_value))
+        x_exit = True
+    else:
+        logging.info('--> x-coordinate: different dist -- p-value: {0:.4f}'.
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_I/01_method_I.py]: logging.error

File: systemtest/jpsreport_tests/method_I/01_method_I.py [hunk 25]
```diff
@@ -0,0 +1,123 @@
...
+                     format(x_p_value))
+        x_exit = False
+        
+    if y_p_value > alpha:
+        logging.info('--> y-coordinate: same dist -- p-value: {0:.4f}'.
+                     format(y_p_value))
+        y_exit = True
+    else:
+        logging.info('--> y-coordinate: different dist -- p-value: {0:.4f}'.
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_I/01_method_I.py]: logging.error

File: systemtest/jpsreport_tests/method_I/01_method_I.py [hunk 26]
```diff
@@ -0,0 +1,123 @@
...
+                     format(y_p_value))
+        y_exit = False
+        
+    if z_p_value > alpha:
+        logging.info('--> z-coordinate: same dist -- p-value: {0:.4f}'.
+                     format(z_p_value))
+        z_exit = True
+    else:
+        logging.info('--> z-coordinate: different dist -- p-value: {0:.4f}'.
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_I/01_method_I.py]: logging.error

File: systemtest/jpsreport_tests/method_I/01_method_I.py [hunk 27]
```diff
@@ -0,0 +1,123 @@
...
+                     format(z_p_value))
+        z_exit = False
+        
+    if density_p_value > alpha:
+        logging.info('--> density: same dist -- p-value: {0:.4f}'.
+                     format(density_p_value))
+        density_exit = True
+    else:
+        logging.info('--> density: different dist -- p-value: {0:.4f}'.
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_I/01_method_I.py]: logging.error

File: systemtest/jpsreport_tests/method_I/01_method_I.py [hunk 28]
```diff
@@ -0,0 +1,123 @@
...
+                     format(density_p_value))
+        density_exit = False
+
+    if velocity_p_value > alpha:
+        logging.info('--> velocity: same dist -- p-value: {0:.4f}'.
+                     format(velocity_p_value))
+        velocity_exit = True
+    else:
+        logging.info('--> velocity: different dist -- p-value: {0:.4f}'.
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_I/01_method_I.py]: logging.error

File: systemtest/jpsreport_tests/method_I/01_method_I.py [hunk 29]
```diff
@@ -0,0 +1,123 @@
...
+                     format(velocity_p_value))
+        velocity_exit = False
+
+
+    success = (Frame_exit == True) and (PersID_exit == True) and \
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_I/01_method_I.py]: see above, check for ==True not needed

File: systemtest/jpsreport_tests/method_I/01_method_I.py [hunk 30]
```diff
@@ -0,0 +1,123 @@
...
+    (x_exit == True) and (y_exit == True) and \
+     (z_exit == True) and (velocity_exit == True) and \
+     (density_exit == True)
+    if not success:
+        logging.info("%s exits with FAILURE" % (argv[0]))
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_I/01_method_I.py]: logging.error

File: systemtest/jpsreport_tests/method_J/01_method_J.py [hunk 31]
```diff
@@ -0,0 +1,59 @@
...
+
+    density_ks_statistic, density_p_value = ks_2samp(data_2[:, 1], data_1[:, 1])
+
+    velocity_ks_statistic, velocity_p_value = ks_2samp(data_2[:, 2], data_1[:, 2])
+
+    if density_p_value > alpha:
+        logging.info('--> density: same dist -- p-value: {0:.4f}'.format(density_p_value))
+        density_exit = True
+    else:
+        logging.info('--> density: different dist -- p-value: {0:.4f}'.format(density_p_value))
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_J/01_method_J.py]: logging.error

File: systemtest/jpsreport_tests/method_J/01_method_J.py [hunk 32]
```diff
@@ -0,0 +1,59 @@
...
+        density_exit = False
+
+    if velocity_p_value > alpha:
+        logging.info('--> velocity: same dist -- p-value: {0:.4f}'.format(velocity_p_value))
+        velocity_exit = True
+    else:
+        logging.info('--> velocity: different dist -- p-value: {0:.4f}'.format(velocity_p_value))
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_J/01_method_J.py]: logging.error

File: systemtest/jpsreport_tests/method_J/01_method_J.py [hunk 33]
```diff
@@ -0,0 +1,59 @@
...
+        velocity_exit = False
+
+    success = (velocity_exit == True) and (density_exit == True)
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_J/01_method_J.py]: check for ==True not needed

File: systemtest/jpsreport_tests/method_J/01_method_J.py [hunk 34]
```diff
@@ -0,0 +1,59 @@
...
+    if not success:
+        logging.info("%s exits with FAILURE" % (argv[0]))
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_J/01_method_J.py]: logging.error

File: systemtest/jpsreport_tests/method_C/01_method_C.py [hunk 35]
```diff
@@ -0,0 +1,58 @@
...
+
+    alpha = 0.01
+    density_ks_statistic, density_p_value = ks_2samp(data_2[:, 1], data_1[:, 1])
+    velocity_ks_statistic, velocity_p_value = ks_2samp(data_2[:, 2], data_1[:, 2])
+
+    if velocity_p_value > alpha:
+        logging.info('--> velocity: same dist -- p-value: {0:.4f}'.format(velocity_p_value))
+        velocity_exit = True
+    else:
+        logging.info('--> velocity: different dist -- p-value: {0:.4f}'.format(velocity_p_value))
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_C/01_method_C.py]: logging.error

File: systemtest/jpsreport_tests/method_C/01_method_C.py [hunk 36]
```diff
@@ -0,0 +1,58 @@
...
+        velocity_exit = False
+
+    if density_p_value > alpha:
+        logging.info('--> density: same dist -- p-value: {0:.4f}'.format(density_p_value))
+        density_exit = True
+    else:
+        logging.info('--> density: different dist -- p-value: {0:.4f}'.format(density_p_value))
```
Comment (schroedtert) [inline systemtest/jpsreport_tests/method_C/01_method_C.py]: logging.error

File: systemtest/jpsreport_tests/method_A/01_method_A.py [hunk 37]
```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+#####################################################################
+# For every source, check  if agents are generated within the bounding box defined
+# with xmin, x_max, y_min, y_max
+#
+# Note:
+# In this test N_Create == agents_max
+# To add new cases increment the group_id by 1.
+# source_id == group_id
```
[the remark this answers is not in the document]
Comment (chraibi) [reply systemtest/jpsreport_tests/method_A/01_method_A.py]: done

File: systemtest/jpsreport_tests/method_A/01_method_A.py (hunk 9 above)
Comment (chraibi) [reply systemtest/jpsreport_tests/method_A/01_method_A.py]: done.

File: systemtest/jpsreport_tests/method_A/01_method_A.py (hunk 1 above)
Comment (chraibi) [reply systemtest/jpsreport_tests/method_A/01_method_A.py]: I'll take critical, okay?

File: systemtest/jpsreport_tests/method_A/01_method_A.py (hunk 2 above)
Comment (chraibi) [reply systemtest/jpsreport_tests/method_A/01_method_A.py]: thanks.

File: systemtest/jpsreport_tests/method_A/01_method_A.py (hunk 3 above)
Comment (chraibi) [reply systemtest/jpsreport_tests/method_A/01_method_A.py]: critical?

File: systemtest/jpsreport_tests/method_B/01_method_B.py (hunk 4 above)
Comment (chraibi) [reply systemtest/jpsreport_tests/method_B/01_method_B.py]: done

**Pull request merged**